src/main/java/com/globant/carrito/cart/Carts.java

```diff
@@ -0,0 +1,134 @@
+package com.globant.carrito.cart;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.Date;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToOne;
+
+import com.globant.carrito.product.Items;
+
+
+@Entity
+public class Carts {
+	
+	@Id
+	@GeneratedValue
+	private int cartId;
+	
+	@ManyToOne(cascade = CascadeType.ALL)
+	private int clientId;
+	
+	@OneToOne(cascade = CascadeType.ALL)
+	private Set<Items> items;
+	
+	@OneToOne(cascade = CascadeType.ALL)

```

Comment:
Ah ya entendí. Pasó que, cuando decidí probar con un enum, ya había creado la relación OneToOne con paymentId y me olvidé de borrarla... Ahí borré la relación al id que no existe:

    @OneToOne(cascade = CascadeType.ALL)
    private int paymentId;


Date: Wed, 4 Feb 2015 15:36:25 -0800
From: notifications@github.com
To: cart_client@noreply.github.com
CC: marcosadriandalmasso@hotmail.com
Subject: Re: [cart_client] Avance del shopping cart hasta el 03/02/15. (#1)

In src/main/java/com/globant/carrito/cart/Carts.java:

> +
> +
> +@Entity
> +public class Carts {
> +	
> +	@Id
> +	@GeneratedValue
> +	private int cartId;
> +	
> +	@ManyToOne(cascade = CascadeType.ALL)
> +	private int clientId;
> +	
> +	@OneToOne(cascade = CascadeType.ALL)
> +	private Set<Items> items;
> +	
> +	@OneToOne(cascade = CascadeType.ALL)

Claro, pero para qué sería el campo private int paymentId; ? Ya tenés private PaymentTypes paymentType;. Mañana si querés probamos como es el tema con las enumeraciones en JPA porque la verdad creo que no las usé nunca (o no me acuerdo, jeje).


—
Reply to this email directly or view it on GitHub.